yt_dlp/extractor/niconicochannelplus.py

```diff
@@ -0,0 +1,455 @@
+import functools
+import json
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    OnDemandPagedList,
+    UserNotLive,
+    filter_dict,
+    int_or_none,
+    parse_iso8601,
+    parse_qs,
+    traverse_obj,
+)
+
+
+class NiconicoChannelPlusBaseIE(InfoExtractor):
+    _WEBPAGE_BASE_URL = 'https://nicochannel.jp'
+
+    def _call_api(self, path, item_id, *args, **kwargs):
+        return self._download_json(
+            f'https://nfc-api.nicochannel.jp/fc/{path}', video_id=item_id, *args, **kwargs)
+
+    def _find_fanclub_site_id(self, channel_name):
+        fanclub_list_json = self._call_api(
+            'content_providers/channels', item_id=f'channels/{channel_name}',
+            note='Fetching channel list', errnote='Unable to fetch channel list',
+        )['data']['content_providers']
+
+        for fanclub in fanclub_list_json:
+            if fanclub.get('domain') == f'{self._WEBPAGE_BASE_URL}/{channel_name}':
+                return fanclub['id']
+
+        raise ExtractorError(f'Non-existing channel: {channel_name}', expected=True)
+
+    def _get_channel_info(self, fanclub_site_id):
+        return self._call_api(
+            f'fanclub_sites/{fanclub_site_id}/page_base_info', item_id=f'fanclub_sites/{fanclub_site_id}',
+            note='Fetching channel info', errnote='Unable to fetch channel info',
+        )['data']['fanclub_site']
+
+
+class NiconicoChannelPlusIE(NiconicoChannelPlusBaseIE):
```

Comment:
[NiconicoChannel (if an IE is named like this)](https://ch.nicovideo.jp/) and niconico channel plus are distinct site (though owned by the same company)

And since the sum of LOC of 2 files (excluding my NNL PR) are over 1k, IMO it's not good to do that